test/Interop/Cxx/class/nonescapable-lifetimebound.swift

```diff
@@ -0,0 +1,63 @@
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: %target-swift-frontend -typecheck -I %swift_src_root/lib/ClangImporter/SwiftBridging  -I %t/Inputs  %t/test.swift -enable-experimental-feature NonescapableTypes -cxx-interoperability-mode=default -diagnostic-style llvm 2>&1
+
+//--- Inputs/module.modulemap
+module Test {
+    header "nonescapable.h"
+    requires cplusplus
+}
+
+//--- Inputs/nonescapable.h
+#include "swift/bridging"
+
+struct SWIFT_NONESCAPABLE View {
+    View() : member(nullptr) {}
+    View(const int *p [[clang::lifetimebound]]) : member(p) {}
+    View(const View&) = default;
+private:
+    const int *member;
+};
+
+struct Owner {
+    int data;
+
+    View handOutView() const [[clang::lifetimebound]] {
+        return View(&data);
+    }
+};
+
+Owner makeOwner() {
+    return Owner{42};
+}
+
+View getView(const Owner& owner [[clang::lifetimebound]]) {
+    return View(&owner.data);
+}
+
+View getViewFromFirst(const Owner& owner [[clang::lifetimebound]], const Owner& owner2) {
+    return View(&owner.data);
+}
+
+bool coinFlip;
+
+View getViewFromEither(const Owner& owner [[clang::lifetimebound]], const Owner& owner2 [[clang::lifetimebound]]) {
+    if (coinFlip)
+        return View(&owner.data);
+    else
+        return View(&owner2.data);
+}
+
+//--- test.swift
+
```

Comment:
Could you please add a module interface test that would check that the Swift lifetime attributes are printed in the generated interface for a C++ module?

See `test/Interop/Cxx/class/access-specifiers-module-interface.swift` for example

Reply:
Unfortunately, the Swift lifetimes are not supported by the module printer yet as there is no official/final syntax for it. That being said, I added checks like that because it is still useful to see what parameters are imported as borrowing. Let me know if that looks good.

Reply:
@meg-gupta is planning to add `@lifetime()` annotations to the module printer.
rdar://130342167 ([nonescapable] add .swiftinterface support for lifetime dependence)

Reply:
You can write an `-emit-sil` test like `test/SIL/lifetime_dependence_buffer_view_test.swift`  to ensure that the SIL representation has `_scope`/ `_inherit` in the interim.

Reply:
A slightly unrelated comment: I might be missing something but `lifetime_dependence_buffer_view_test.swift` does not seem to invoke `FileCheck`? Is that intentional, or a bug?

Reply:
Thanks. I'll fix `lifetime_dependence_buffer_view_test.swift` to include FileCheck!